src/lib.rs

```diff
@@ -1484,6 +1487,25 @@ pub fn shallow_scan_for_device(
         .find(|&device| device.vendor_id == vendor_id && device.device_id == device_id)
 }
 
+// same as shallow_scan_for_device but scans all instances, some devices like
+// RoT have multiple instances within each chiplet, we should find and return all.
```

Comment:
This is a generic library, so let's avoid any specific examples for any design here.

Reply:
done